client/client.go

```diff
@@ -144,6 +147,24 @@ func New(cfg Config) (Client, error) {
 	return c, nil
 }
 
+// KeepSync keeps synchronizing with etcd cluster about its member list,
+// so client could update its endpoints and catch up the change of the cluster.
+func KeepSync(ctx context.Context, c Client) error {
```

Comment:
> The only reason http.Serve blocks is that you can consume the returned error.

ok. i could try this idea.

The concern is that AutoSync may return err when single sync action fails today. This gives caller the chance to log in its own way, and notices the caller that the client may be not in sync in the last interval. It could introduce logger and unhealthy channel for this, but it may become complicated.